src/appleseed/foundation/image/exrimagefilewriter.cpp

```diff
@@ -68,26 +68,35 @@ namespace foundation
 // EXRImageFileWriter class implementation.
 //
 
-namespace
+void EXRImageFileWriter::write(
+    const char*             filename,
+    const ICanvas&          image,
+    const ImageAttributes&  image_attributes)
 {
-    const char* ChannelName[] = { "R", "G", "B", "A" };
+    static const char* ChannelNames[] = {"R", "G", "B", "A"};
+    const CanvasProperties& props = image.properties();
+    write(filename, image, image_attributes, props.m_channel_count, ChannelNames);
 }
 
 void EXRImageFileWriter::write(
     const char*             filename,
     const ICanvas&          image,
-    const ImageAttributes&  image_attributes)
+    const ImageAttributes&  image_attributes,
+    const size_t            channel_count,
+    const char**            channel_names)
 {
+    assert(channel_names);
+
     initialize_openexr();
 
     try
     {
+        // todo: lift this limitation.
+        assert(channel_count <= 4);
```

Comment:
`props.m_channel_count` is still used in this method (to compute the X stride). I think it's correct, but just checking.